### PR TITLE
Update supported version documentation

### PR DESCRIPTION
**Pull request by Frassle**

### CODE REVIEW

File: themes/default/content/docs/intro/languages/dotnet.md [hunk 1]
```diff
@@ -12,7 +12,9 @@ aliases: ["/dotnet/"]
 
 <img src="/logos/tech/dotnet.png" align="right" width="150" style="padding:8px; margin-top: -64px">
 
-Pulumi supports infrastructure as code using any .NET language. You can use your favorite .NET tools &mdash; such as editors, package managers, build systems, and test frameworks &mdash; to create, deploy, and manage infrastructure on any cloud, including Azure, AWS, and Google Cloud.
+Pulumi supports writing your infrastructure as code in any .NET language programs running on dotnet using any [Supported version](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#lifecycle).
```
Comment (squaremo) [inline themes/default/content/docs/intro/languages/dotnet.md:15]:
```suggestion
Pulumi supports writing your infrastructure as code in any .NET language programs running on dotnet using any [supported version](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#lifecycle).
```

Here and similarly below: this wouldn't usually be capitalised, and it doesn't look like the term "supported version" is given any special treatment on the linked pages. The word "Support" does appear capitalised in product names (e.g., "Oracle Premier Support"), and when it's the first word in a sentence or heading -- but not in general, and not within the term "supported versions".

File: themes/default/content/docs/intro/languages/go.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ menu:
 
 <img src="/logos/tech/logo-golang.png" align="right" width="150" style="padding:8px; margin-top: -64px">
 
-Pulumi supports writing your infrastructure as code using the Go language. Go 1.16 or later is required.
+Pulumi supports writing your infrastructure as code in the Go language built for any [Supported versions](https://go.dev/doc/devel/release#policy).
```
Comment (squaremo) [inline themes/default/content/docs/intro/languages/go.md:13]:
```suggestion
Pulumi supports writing your infrastructure as code in the Go language built for any [supported versions](https://go.dev/doc/devel/release#policy).
```

File: themes/default/content/docs/intro/languages/java.md [hunk 3]
```diff
@@ -10,7 +10,9 @@ menu:
 
 <img src="/logos/tech/java.svg" align="right" width="90" style="padding:8px; margin-top: -64px">
 
-Pulumi supports writing your infrastructure as code using the Java language. Java 11 or later is required. We support running Maven Pulumi programs, Gradle Pulumi programs, and Pulumi programs packaged as JAR files.
+Pulumi supports writing your infrastructure as code using the Java language running on any [Supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11.
```
Comment (squaremo) [inline themes/default/content/docs/intro/languages/java.md:13]:
```suggestion
Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11.
```

File: themes/default/content/docs/intro/languages/javascript.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ aliases: ["/docs/reference/javascript/"]
 
 <img src="/logos/tech/logo-nodejs.png" align="right" width="150" style="padding:8px; margin-top: -64px">
 
-Pulumi supports JavaScript programs running on Node.js using any of the [Current, Active and Maintenance LTS versions](https://nodejs.org/en/about/releases/).
+Pulumi supports writing your infrastructure as code in any JavaScript language running on Node.js using any of the [Current, Active and Maintenance LTS versions](https://nodejs.org/en/about/releases/).
```
Comment (squaremo) [inline themes/default/content/docs/intro/languages/javascript.md:15]: (these _are_ proper nouns, as they are used like product names in the linked page)

File: themes/default/content/docs/intro/languages/python.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ aliases: ["/docs/reference/python/"]
 
 <img src="/logos/tech/logo-python.svg" align="right" width="150" style="padding:8px; margin-top: -64px">
 
-Pulumi supports programs written in Python 3.6 or later.
+Pulumi supports writing your infrastructure as code in the Python language running on any [Supported version](https://devguide.python.org/versions/#versions).
```
Comment (squaremo) [inline themes/default/content/docs/intro/languages/python.md:15]:
```suggestion
Pulumi supports writing your infrastructure as code in the Python language running on any [supported version](https://devguide.python.org/versions/#versions).
```

File: themes/default/content/docs/intro/languages/javascript.md (hunk 4 above)
Comment (susanev) [reply themes/default/content/docs/intro/languages/javascript.md:15]: we could prolly drop the casing on our page though

**Review by susanev — APPROVED** [review]
thank you! looks great, though i agree with Michaels comments so might be good to adjust the casing in a few situations.

File: themes/default/content/docs/intro/languages/java.md [hunk 6]
```diff
@@ -10,7 +10,9 @@ menu:
...
+
+We support running Maven Pulumi programs, Gradle Pulumi programs, and Pulumi programs packaged as JAR files.
```
Comment (t0yv0) [inline themes/default/content/docs/intro/languages/java.md:15]: Great start. FWIW Maven and Gradle have documented lower bounds https://github.com/pulumi/pulumi-java#requirements  - but I don't think we figured out the LTS strategy yet. We can refine this later. LGTM.

File: themes/default/content/docs/intro/languages/go.md (hunk 2 above)
Comment (cnunciato) [reply themes/default/content/docs/intro/languages/go.md:13]: I added a tweak to @squaremo's suggestion to make `support version` singular (i.e., "for any any supported version").

File: themes/default/content/docs/intro/languages/java.md (hunk 3 above)
Comment (cnunciato) [reply themes/default/content/docs/intro/languages/java.md:13]: "After 11" reads a bit ambiguously to me. (My first thought was, _Oh, I thought we supported 11 also. Do we only support 12 and up?_) If we do mean "11 or later", we should probably say that.

**Review by cnunciato — APPROVED** [review]
Agree with all of the suggestions here. Thanks for doing this!

File: themes/default/content/docs/intro/languages/java.md (hunk 3 above)
Comment (Frassle) [reply themes/default/content/docs/intro/languages/java.md:13]: It's 11 or later. So "Pulumi supports writing your infrastructure as code using the Java language running on any supported version after Java 11 or later." ?

File: themes/default/content/docs/intro/languages/java.md [hunk 7]
```diff
@@ -10,7 +10,9 @@ menu:
 
 <img src="/logos/tech/java.svg" align="right" width="90" style="padding:8px; margin-top: -64px">
 
-Pulumi supports writing your infrastructure as code using the Java language. Java 11 or later is required. We support running Maven Pulumi programs, Gradle Pulumi programs, and Pulumi programs packaged as JAR files.
+Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11.
```
Comment (Frassle) [inline themes/default/content/docs/intro/languages/java.md:13]:
```suggestion
Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11 or later.
```

Comment (Frassle) [reply themes/default/content/docs/intro/languages/java.md:13]: Does this read ok?

File: themes/default/content/docs/intro/languages/javascript.md (hunk 4 above)
Comment (RobbieMcKinstry) [reply themes/default/content/docs/intro/languages/javascript.md:15]: I don't understand what "any JavaScript language" means. There is only one JavaScript language, even if there are different reference implementations.

Comment (Frassle) [reply themes/default/content/docs/intro/languages/javascript.md:15]: yeh I was trying to succinctly cover JavaScript/TypeScript/anything else that transpiled.

File: themes/default/content/docs/intro/languages/java.md (hunk 7 above)
Comment (cnunciato) [inline themes/default/content/docs/intro/languages/java.md:13]: `after` was the problematic word, so I might just suggest `of` instead:

```suggestion
Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) of Java 11 or later.
```

What do you think?

**Pull request merged**